File: languages/go/entity/availability/ipfs_publisher.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IpfsPublisher represents a publisher for IPFS data availability.
type IpfsPublisher struct {
	Type proto.DataAvailabilityType
	Args PublisherArgs
}

// IpfsPublisher represents a publisher for IPFS data availability.
func NewIpfsPublisher() IpfsPublisher {
	return IpfsPublisher{
		Type: proto.DataAvailabilityType_IPFS,
		Args: PublisherArgs{},
	}
}

func (e IpfsPublisher) ToProto() *proto.Publisher {
	return &proto.Publisher{
		Type: e.Type,
		Args: e.Args.ToProto(),
	}
}

File: languages/js/src/entity/availability/ipfs_publisher.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { Publisher } from "./publisher";
import { PublisherArgs } from "./publisher_args";

/**
 * Represents a publisher for IPFS data availability.
 */
export class IpfsPublisher implements Publisher {
  type: proto.DataAvailabilityType;
  args: PublisherArgs;

  /**
   * Constructs a IpfsPublisher object with the specified parameters.
   */
  constructor() {
    this.type = proto.DataAvailabilityType.IPFS;
    this.args = new PublisherArgs();
  }

  public toProto(): proto.Publisher {
    return proto.Publisher.fromPartial({
      type: this.type,
      args: this.args.toProto()
    });
  }
}
